src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/single_op_graph.cpp

```diff
@@ -741,6 +741,15 @@ std::shared_ptr<ov::Model> generate(const std::shared_ptr<ov::op::v1::Pad> &node
     return std::make_shared<ov::Model>(results, params, "Pad-1");
 }
 
+std::shared_ptr<ov::Model> generate(const std::shared_ptr<ov::op::v12::Pad> &node) {
+    const auto params = ngraph::builder::makeDynamicParams(ov::element::f32, {{6, 10}});
+    const auto pad_begin = ngraph::builder::makeConstant<int64_t>(ov::element::i64, {2}, {4, -2});
+    const auto pad_end = ngraph::builder::makeConstant<int64_t>(ov::element::i64, {2}, {5, -1});
+    const auto pad = std::make_shared<ov::op::v12::Pad>(params[0], pad_begin, pad_end, ov::op::PadMode::CONSTANT);
+    ov::ResultVector results{std::make_shared<ov::op::v0::Result>(pad)};
+    return std::make_shared<ov::Model>(results, params, "Pad-12");
+}
+
```

Comment:
May be it makes sense to check a case with negative `pad_begin` and positive `pad_end` value for the same dimension, i.e. on the one corner it is cropped and on the opposite the image is padded.